linux/tests/threadpool_linux_ut/threadpool_linux_ut.c

```diff
@@ -1140,9 +1141,10 @@ TEST_FUNCTION(threadpool_timer_start_with_NULL_work_function_context_succeeds)
     // arrange
     THANDLE(THREADPOOL) threadpool = test_create_and_open_threadpool();
 
-    TIMER_INSTANCE_HANDLE timer_instance;
+    THANDLE(TIMER_INSTANCE) timer_instance = NULL;
```

Comment:
>timer_instance = NULL [](http://example.com/codeflow?start=28&length=21)

Do we have to assign this to be NULL for `threadpool_timer_start` to work? I can see this might be an annoyance when the timer is part of a struct, like it generally is.

Reply:
will getting segmentation fault if not set this THANDLE to NULL

Reply:
The common pattern we have in the code that uses this is
```c
typedef struct SOME_STRUCT_TAG
{
//  ...
THANDLE(TIMER_INSTANCE) timer;
// ...
} SOME_STRUCT;

// then in the "_create" function:
SOME_STRUCT_HANDLE  result = (SOME_STRUCT_HANDLE)malloc(sizeof(SOME_STRUCT);
threadpool_timer_start(threadpool, start_delay, period, test_work_function, context, &result->timer);
```

The timer is being passed in uninitialized, and so requiring it to be initialized in the test implies that the above pattern will need to change. BTW, this comment goes along with using `THANDLE_INITIALIZE_MOVE` instead of `THANDLE_MOVE`

Reply:
modified to initialize_move